### ISSUE TITLE
IngestArch: Collect (and incorporate) feedback on Phase 1 architectures

### ISSUE THREAD

**Issue by karenzone** [elastic/ingest-docs#56]
- [ ] [Elastic Agent to Elasticsearch](https://github.com/elastic/ingest-docs/edit/main/docs/en/ingest-arch/1-agent-to-es.asciidoc)

     - [ ] [Elastic Agent to Elasticsearch: Agent installed](https://github.com/elastic/ingest-docs/edit/main/docs/en/ingest-arch/1a-agent.asciidoc)
     - [ ] [Elastic Agent to Elasticsearch: APIs for collection](https://www.elastic.co/guide/en/ingest/current/agent-apis.html)

- [ ] [Elastic Agent to proxy to Elasticsearch](https://www.elastic.co/guide/en/ingest/current/agent-proxy.html)

- [ ] [Elastic Agent to Logstash to Elasticsearch](https://www.elastic.co/guide/en/ingest/current/agent-ls.html)
     - [ ] [Elastic Agent to Logstash (for enrichment) to Elasticsearch](https://www.elastic.co/guide/en/ingest/current/ls-enrich.html)
     - [ ] [Elastic Agent to Logstash to Elasticsearch: Logstash Persistent Queue (PQ) for buffering](https://www.elastic.co/guide/en/ingest/current/lspq.html)
     - [ ] [Elastic Agent to Logstash to Elasticsearch: Logstash to bridge networks](https://www.elastic.co/guide/en/ingest/current/ls-networkbridge.html)
     - [ ] [Elastic Agent to Logstash for routing to multiple Elasticsearch clusters and additional destinations](https://www.elastic.co/guide/en/ingest/current/ls-multi.html)

- [ ] [Elastic Agent to Elasticsearch with Kafka as middleware message queue](https://www.elastic.co/guide/en/ingest/current/agent-kafka-es.html)
     - [ ] [Elastic Agent to Logstash to Kafka to Kafka ES Sink to Elasticsearch: Kafka as middleware message queue](https://www.elastic.co/guide/en/ingest/current/agent-kafka-essink.html)
     - [ ] [Elastic Agent to Logstash to Kafka to Logstash to Elasticsearch: Kafka as middleware message queue](https://www.elastic.co/guide/en/ingest/current/agent-kafka-ls.html)

- [ ] [Logstash to Elasticsearch](https://www.elastic.co/guide/en/ingest/current/ls-for-input.html)

**Comment by karenzone** [comment]
Status: I've made a first pass and called out items that need attention.

**Comment by karenzone** [comment]
@kpollich @cmacknz @joshdover @jsvd

**Comment by amitkanfer** [comment]
Took a first stub at the docs, thank you @karenzone ! Attaching my comments, happy to jump on a call / review again when available. @mukeshelastic @nimarezainia is this before or after your review?

- In this [doc](https://github.com/elastic/ingest-docs/edit/main/docs/en/ingest-arch/1a-agent.asciidoc), not sure what it means to say `(agent installed)` (in the ingest model->data path section)
- `An Elastic Agent integration exists in the Kibana integrations UI for software components on which you can install Elastic Agent`, one can understand that if he wants to ingest nginx logs, the elastic agent should be installed on the nginx "software component", which is not the case. I would rephrase it to something more simple like "An Elastic Agent integration exists in the Kibana integrations UI for the software you want to monitor / observe / protect.
- Examples list is still empty. Why? Should be relatively easy to fetch from our telemetry.
- In [this](https://www.elastic.co/guide/en/ingest/current/agent-proxy.html) doc i would make it clear in the diagram what's the control path and what's the data path.
- in the same [document](https://www.elastic.co/guide/en/ingest/current/agent-proxy.html), the control and data paths are mixed. Fleet server is not in the data path.
- same document. typo: `preventconnection`
- I would love if @cmacknz can double check what's written [here](https://www.elastic.co/guide/en/fleet/master//fleet-agent-proxy-support.html) on proxy configuration
- do we also want to say something about client certs and mTLS?
- well done on the logstash use cases!
- On the enrichment [doc](https://www.elastic.co/guide/en/ingest/current/ls-enrich.html), i would link to a few enrich processors we have (there's a link at the bottom to "Transform" which is a bit different than just "enrich")
- "Data sources that change frequently, such as stock ticker data" - why is this an example to use Logstash? does it imply that Agent is not fast enough? or is it because the enriched data is "public" data source and thus we need Logstash's custom HTTP/tcp input plugin?
- in [this](https://www.elastic.co/guide/en/ingest/current/ls-enrich.html#ls-enrich-proc) process overview, worth adding how to configure logstash as an output to Agent? (in the agent policy UI)
- Same thing for all 4 Agent to LS documents
- We have some PQ performance benchmarks today. @roaksoax should we link it [here](https://www.elastic.co/guide/en/ingest/current/lspq.html)?
- Worth adding a visual in the [diagram](https://www.elastic.co/guide/en/ingest/current/ls-networkbridge.html) to show the two different networks that LS is used to bridge between.
- "Use when"... the same explanation was given to put a proxy between agents and ES, so what makes this use case unique to use Logstash?
- Examples are missing...
- we will soon need to update [this](https://www.elastic.co/guide/en/ingest/current/ls-for-input.html) doc because LS can be deployed on k8s as well (ECK). I would just avoid specifying the host for the logstash instance, not sure what value it adds.

- (relevant to all docs) Can we add an animated gif to show the installation flow? perhaps a link to a webinar / youtube video?

**Comment by karenzone** [comment]
Feedback from @joshdover:
> are we planning to have an overview doc that explains which section to look at based on the use case? It feels like the "use when" sections are buried too deep. I imagine a user approaching these docs with the "use when" already in mind and we want to guide them from that entry point to the correct section.
> I think it will be important for the first release of this.

**Comment by mukeshelastic** [comment]
@amitkanfer the current state of the docs is after number of working sessions and reviews with @karenzone. The requested review here is to address comments from last review. will take a pass at it later today. Thanks @karenzone

**Comment by mukeshelastic** [comment]
> are we planning to have an overview doc that explains which section to look at based on the use case? It feels like the "use when" sections are buried too deep. I imagine a user approaching these docs with the "use when" already in mind and we want to guide them from that entry point to the correct section.
> I think it will be important for the first release of this.

When we wrote a [google doc](https://docs.google.com/document/d/1u2uONewPiuj4jL9VCnQa2auBuES8iKqtJ5q_Ox5jYSU/edit#) on the architecture requirements, we took the approach of describe a use case and follow up with few architecture choices. It because a little bit clumsy because for each use case, we will likely have more than one architecture.. So somewhere along the line, we decided to switch to describe an architecture and include a "use when" section to communicate the use case the architecture addresses.  So my suggestion is ship this arch diagrams as is now and depending on the feedback, we can provide an alternative organization.

**Comment by mukeshelastic** [comment]
@karenzone addressed the items you left for me in the issue. Amit left good feedback, can you create the line items for his feedback just like how you did for my feedback in the issue description and then we can collectively decide which of his comments to address and which one to leave for future.

**Comment by cmacknz** [comment]
> I would love if @cmacknz can double check what's written [here](https://www.elastic.co/guide/en/fleet/master//fleet-agent-proxy-support.html) on proxy configuration

It's correct, the only thing we should we should note is that if the user a needs a proxy specifically for connects to artifacts.elastic.co (separate from the Fleet Server proxy and the Elasticsearch proxy) they can configure it by setting the proxy URL in [Agent Binary Download Settings](https://www.elastic.co/guide/en/fleet/master//fleet-settings.html#fleet-agent-binary-download-settings).

**Comment by karenzone** [comment]
Thank you @mukeshelastic, @amitkanfer, @joshdover, @cmacknz for the great questions, answers, and feedback. I'll make sense of this all and rein it in (as @mukeshelastic suggested) for the next sprint. 

If we're all agreement that we have a good set of actionable feedback, I'll close this issue as Done in Sprint 8. We'll pick up in Sprint 9 with quantifying, prioritizing, and implementing feedback. 

If you're worried that this is your last change to provide feedback, don't be!  Our "docs as code" model, github, and doc builds every half hour mean that we are continually iterating.

**Comment by karenzone** [comment]
> We took the approach of describe a use case and follow up with few architecture choices. It because a little bit clumsy because for each use case, we will likely have more than one architecture.. So somewhere along the line, we decided to switch to describe an architecture and include a "use when" section to communicate the use case the architecture addresses. So my suggestion is ship this arch diagrams as is now and depending on the feedback, we can provide an alternative organization.

It's tricky for sure because it's not always a one-to-one relationship.  The one-to-many angle made for some interesting doc architecture challenges.

**Comment by karenzone** [comment]
Thank you all.  I'm closing this issue for S8, and moving the comments into a new meta issue for S9. 
No need to worry! You'll have more opportunities to provide feedback as we continue to iterate.
